frontend/packages/metal3-plugin/src/components/types.ts

```diff
@@ -10,7 +10,8 @@ export type StatusProps = {
   status: string;
   title?: string;
   description?: string;
-  [key: string]: any;
+  maintenance?: K8sResourceKind;
+  className?: string;
```

Comment:
I wouldn't add these two, this defines general shape of the status where it has status, title, description and a set of optional props which are provided as needed. These can be further specified in 'BareMetalHostStatusProps' and 'BareMetalNodeStatusProps'